sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -263,4 +280,107 @@ case class InsertIntoHadoopFsRelationCommand(
       }
     }.toMap
   }
+
+  /**
+   * Detect the conflict when there are several InsertIntoHadoopFsRelation operations
+   * write to same partition in the same table or a non-partitioned table concurrently.
+   */
+  private def detectConflict(
+      fs: FileSystem,
+      path: Path,
+      staticPartitionKVs: Seq[(String, String)]): Unit = {
+    for (i <- 0 until partitionColumns.size) {
```

Comment:
The scaladoc mentions non-partitioned tables; won't `partitionColumns.size` be 0 in that case, in which case this will always pass?

Reply:
thanks. my mistake.
It should be `for (i <- 0 until partitionColumns.size + 1)`.